--- freeswitch-1.0.6/src/mod/applications/mod_commands/mod_commands.c	2010-04-07 02:05:28.000000000 +0800
+++ freeswitch-1.0.6/src/mod/applications/mod_commands/mod_commands.c	2012-12-24 12:27:17.000000000 +0800
@@ -990,6 +990,8 @@
 		free(expanded);
 	}
 
+    switch_event_destroy( &event); //chenyuzhen add to free memory
+
 	return SWITCH_STATUS_SUCCESS;
 }
 
@@ -2457,7 +2459,7 @@
 
 	if (!(rsession = switch_core_session_locate(uuid))) {
 		stream->write_function(stream, "-ERR Cannot locate session!\n");
-		return SWITCH_STATUS_SUCCESS;
+        goto done;
 	}
 
 	if (!strcasecmp(action, "start")) {
