--- freeswitch-1.0.6/src/mod/event_handlers/mod_event_socket/mod_event_socket.c	2010-04-07 02:05:28.000000000 +0800
+++ freeswitch-1.0.6/src/mod/event_handlers/mod_event_socket/mod_event_socket.c	2011-10-19 11:19:32.000000000 +0800
@@ -286,12 +286,14 @@
 			}
 		}
 
-		if (send && l->filters && l->filters->headers) {
+		if (send){
+			switch_mutex_lock(l->filter_mutex);
+			
+			if(l->filters && l->filters->headers) {
 			switch_event_header_t *hp;
 			const char *hval;
-
 			send = 0;
-			switch_mutex_lock(l->filter_mutex);
+			
 			for (hp = l->filters->headers; hp; hp = hp->next) {
 				if ((hval = switch_event_get_header(event, hp->name))) {
 					const char *comp_to = hp->value;
@@ -335,6 +337,7 @@
 					}
 				}
 			}
+			}
 			switch_mutex_unlock(l->filter_mutex);
 		}
 
@@ -1278,7 +1281,8 @@
 		}
 
 		if (do_sleep) {
-			switch_cond_next();
+			//switch_cond_next();
+            switch_micro_sleep(50000);
 		}
 	}
 
