--- freeswitch-1.0.6/src/mod/formats/mod_sndfile/mod_sndfile.c	2010-04-07 02:05:28.000000000 +0800
+++ freeswitch-1.0.6/src/mod/formats/mod_sndfile/mod_sndfile.c	2012-04-28 13:34:03.000000000 +0800
@@ -107,7 +107,8 @@
 		if (handle->samplerate == 8000 || handle->samplerate == 16000 ||
 			handle->samplerate == 24000 || handle->samplerate == 32000 || handle->samplerate == 48000 ||
 			handle->samplerate == 11025 || handle->samplerate == 22050 || handle->samplerate == 44100) {
-			context->sfinfo.format |= SF_FORMAT_PCM_16;
+			//context->sfinfo.format |= SF_FORMAT_PCM_16;
+			context->sfinfo.format |= SF_FORMAT_PCM_U8;
 		}
 	}
 
